server-spi-private/src/main/java/org/keycloak/userprofile/Attributes.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ *  * Copyright 2021  Red Hat, Inc. and/or its affiliates
+ *  * and other contributors as indicated by the @author tags.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package org.keycloak.userprofile;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+/**
+ * <p>This interface wraps the attributes associated with a user profile. Different operations are provided to access and
+ * manage these attributes.
+ *
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+public interface Attributes extends Map<String, List<String>> {
```

Comment:
We could remove this interface and move the methods herein to the `UserProfile` itself. The motivation behind this abstraction is to avoid publishing too many operations in `UserProfile`. I can see in the future more operations we can perform on the attributes associated with a profile.

Reply:
We will probably see later as we start to implement configuration parsing and validation. I see methods like isReadonlyAttribute() and validate() here and all this is implemented in DefaultAttributes then.